lib/index.js

```diff
@@ -35,13 +35,23 @@ function traverse(input, strict) {
  */
 
 function object(obj, strict) {
+  // 'each' utility uses obj.length to check whether the obj is array. To avoid incorrect classification, wrap call to 'each' with rename of obj.length
+  if (obj.length && typeof obj.length === 'number' && !(obj.length - 1 in obj)) { // cross browser compatible way of checking has length and is not array
```

Comment:
@tsholmes I think this addresses your concerns, only thing that could break this is if it was a nonarray object with `length` and a property called `length-1` but is that such an improbable edge case that we can ignore it?

Reply:
Yeah I think there's only so much we can do. they could send `{ '0': 'a', '1': 'b', 'length': 2 }` and we just have to assume its an array